LSPS0/common-schemas.md

```diff
@@ -0,0 +1,245 @@
+the big-endian 32-byte integer representation of the X coordinate.
+
+In contexts where the case is significant (for example, if it
+will be committed to by some hash or signature) then the
+representation MUST be in lowercase.
+Otherwise, readers MUST allow both lowercase and uppercase
+hexadecimal digits.
+
+Readers SHOULD validate that the point is indeed on the SECP256K1
+curve.
+
+For example, the SECP256K1 generator point `G` would be written
+as the JSON string
+`"0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798"`.
+
+> **Rationale** Bitcoin and Lightning have standardized on this
+> format, and traditionally used lowercase for hexadecimal digits.
+
+### Lightning Network Connection Strings
+
+Lightning Network connection strings describe a Lightning Network
+Node ID and one way to connect to that node.
+
+Connection strings MUST be encoded as a JSON string, with three
+parts:
+
+1.  The Lightning Network Node ID, encoded as the hexadecimal
+    dump of the compressed DER encoding, followed by the `@`
+    sign.
+2.  The address, which follows the `@` sign and lasts to the
+    end of the string, or the last `:` in the string, or
+    delimited by characters `[` and `]`.
+    Address formats allowed are those defined in the BOLT
+    specifications:
+    * IPv4 addresses.
+    * IPv6 addresses.
+    * A DNS-resolvable name.
+    * A Torv3 hidden service.
+3.  An optional port number, which is started by the last `:`
+    in the string (unless the last `:` is inside `[` and `]`
+    delimiters), and is a decimal text representation of
+    the number.
+    If unspecified, it should be treated as defaulting to
+    port 9735.
+
+For example, a node with ID `0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798`
+on an IPv6 `localhost` listening on the default
+port 9735 might be expressed as the JSON string:
+`"0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798@[::1]"`
+
+> **Rationale** The address may be delimited by explicit `[`
+> and `]` characters since IPv6 addresses may contain `:`, and
+> port number is optional.
+
+### Lightning Network Node Signatures
+
+Signatures generated by a particular Lightning Network node, with a
+particular known node ID, MUST be generated and represented using
+the [LND `signmessage` #specinatweet][], and encoded as a JSON string
+containing the zbase32 encoding.
+
+[LND `signmessage` #specinatweet]: https://web.archive.org/web/20191010011846/https://twitter.com/rusty_twit/status/1182102005914800128
+
+> **Rationale** This non-BOLT specification is widely immplemented
+> and is commonly used in many bespoke protocols, often used to
+> prove that a particular user owns a particular node, or as an
+> informal way to reduce spam by requiring that a participant
+> prove they have a published Lightning Network node.
+
+### Datetimes
+
+Particular points of time in the modern era (a "datetime") MUST be
```

Comment:
I am a fan of suffixes. I am sure though not everybody is. What do you thing of requiring a `_at` suffix for every datetime field?

Reply:
Not a fan of suffixes myself.